response_operations_ui/views/collection_exercise.py

```diff
@@ -79,46 +79,18 @@
 }
 
 
-def build_collection_exercise_details(short_name: str, period: str, include_ci: bool = False) -> dict:
-    """
-    :param short_name: short name of the survey (e.g., MBS, BRES, etc)
-    :param period: Period for the collection exercise
-    :param include_ci: Flag to include collection instrument data or not
-    :return: A dict containing useful data about a given collection exercise
-    """
-    survey = survey_controllers.get_survey_by_shortname(short_name)
-    survey_id = survey["id"]
-    exercises = collection_exercise_controllers.get_collection_exercises_by_survey(survey_id)
-    exercise = get_collection_exercise_by_period(exercises, period)
-    if not exercise:
-        logger.error("Failed to find collection exercise by period", short_name=short_name, period=period)
-        abort(404)
-
-    collection_exercise_id = exercise["id"]
-    survey["shortName"] = format_short_name(survey["shortName"])
-    full_exercise = collection_exercise_controllers.get_collection_exercise_by_id(collection_exercise_id)
-    exercise_events = collection_exercise_controllers.get_collection_exercise_events_by_id(collection_exercise_id)
+def get_sample_summary(collection_exercise_id):
     summary_id = collection_exercise_controllers.get_linked_sample_summary_id(collection_exercise_id)
     sample_summary = sample_controllers.get_sample_summary(summary_id) if summary_id else None
+    return sample_summary
 
-    exercise_dict = {
-        "survey": survey,
-        "collection_exercise": full_exercise,
-        "events": convert_events_to_new_format(exercise_events),
-        "sample_summary": _format_sample_summary(sample_summary),
-    }
 
-    if include_ci:
-        exercise_dict["collection_instruments"] = _build_collection_instruments_details(
-            collection_exercise_id, survey_id
-        )
-        if survey["surveyMode"] in ("EQ_AND_SEFT", "EQ"):
-            exercise_dict["eq_ci_selectors"] = (
-                collection_instrument_controllers.get_collection_instruments_by_classifier(
-                    ci_type="EQ", survey_id=survey_id
-                )
-            )
-    return exercise_dict
+def get_collection_exercise_and_survey_details(short_name, period):
```

Comment:
Every use of build_collection_exercise_details needed to get the survey from the shortname and then find the exercise, which makes complete sense as the UI only uses short name and the CE table only holds the survey uuid. This therefore makes complete sense to have a single function to do this. 

The majority of usages of build_collection_exercise_details needed just the exercise, which just means this can be used. The ones that needed more I then dealt with on demand, moving us away from redundancy